DocumentService/Repositories/DocumentRepository.cs

```diff
@@ -0,0 +1,149 @@
+                return numberOfEntitiesUpdated;
+            }
+            catch (Exception exception)
+            {
+                throw;
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<DocumentInfo> GetDocumentAsync(Guid id)
+        {
+            try
+            {
+                DocumentInfo documentInfo = await this.GetDocument(id);
+                if (documentInfo != null)
+                {
+                    return documentInfo;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<bool> SetFileDeleted(Guid id, string userId)
+        {
+            try
+            {
+                DocumentInfo documentInfo = await this.GetDocument(id);
+
+                if (documentInfo != null)
+                {
+                    documentInfo.IsDeleted = true;
+                    documentInfo.DeletedById = userId;
+                    documentInfo.DateDeleted = DateTime.UtcNow;
+                    this.context.DocumentInfo.Update(documentInfo);
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<bool> Update(DocumentInfo documentInfo)
+        {
+            var updatedDocumentInfo = await this.GetDocument(documentInfo.DocumentId);
+            try
+            {
+                if (updatedDocumentInfo != null)
+                {
+                    updatedDocumentInfo.UserLastUpdatedById = documentInfo.UserLastUpdatedById;
+                    updatedDocumentInfo.FileName = documentInfo.FileName;
+                    updatedDocumentInfo.FileType = documentInfo.FileType;
+                    updatedDocumentInfo.Description = documentInfo.Description;
+                    updatedDocumentInfo.SubmissionMethod = documentInfo.SubmissionMethod;
+                    updatedDocumentInfo.Language = documentInfo.Language;
+                    updatedDocumentInfo.DocumentTypes = documentInfo.DocumentTypes;
+                    updatedDocumentInfo.DateLastUpdated = DateTime.UtcNow;
+
+                    this.context.DocumentInfo.Update(updatedDocumentInfo);
+                    return true;
+
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<DocumentInfo> GetDocumentsByIds(Guid[] ids)
+        {
+            return this.Filter(x => ids.Contains(x.DocumentId));
```

Comment:
ids.Equals() might be best

Reply:
here we're checking if the list contains the DocumentId, Equals would not work in our case.

Reply:
Sounds good